### ISSUE TITLE
ttnn permute producing garbled data in a non-deterministic way

### ISSUE THREAD

**Issue by nsorabaTT** [tenstorrent/tt-metal#21803]
This issue was found during debug of https://github.com/tenstorrent/tt-metal/issues/21438

When transpose dims are the last two and input is a 5D tensor, the MultiCoreTileInvariant version is called for permute. Here is an example:

Input: [1,1,8,64,256] We want to transpose this to 1,1,8,256,64]

With this input, lets run transpose and see:

import torch
import ttnn

def test_transpose(device):
    # Test parameters
    dim1, dim2, dim3, dim4, dim5 = 1, 1, 8, 64, 256
    dtype = ttnn.bfloat8_b

    for _ in range(100):
        torch_input = torch.randn(dim1, dim2, dim3, dim4, dim5, dtype=torch.bfloat16)
        ttnn_input = ttnn.from_torch(torch_input, dtype, layout=ttnn.Layout.TILE, device=device)
        ttnn_output = ttnn.permute(ttnn_input, [0, 1, 2, 4, 3])

        torch_output = ttnn.to_torch(ttnn_output, dtype=torch.bfloat16)

        if torch.any(torch.isnan(torch_input)) or torch.any(torch.isinf(torch_input)):
            return

        if (torch.any(torch.isinf(torch_output)) or (torch.any(torch.isnan(torch_output)))):
            # Save the torch_input tensor to a file
            torch.save(torch_input, 'tensor.torch')
            assert False, "Found infinity values in ttnn but not in pytorch"
TT_METAL_WATCHER=120 pytest snippet.py

With this, we can see the values are garbled and some values are flushed to infinity. Once the garbled values are introduced, they propagate downstream before ending up as PCC errors in our unit tests.

Note that when tensor rank is 4 or lower or when dimensions to permute are not the last two, the specific permute kernel is not called, hence we see accuracy issues.

**Comment by nsorabaTT** [comment]
Hi @ntarafdar, @sjameelTT requested I assign this to you for triage and scheduling. Thank you for looking into this!

**Comment by sjameelTT** [comment]
#21927 looks like it's an issue after transpose_wh_tile is called. Filed the issue to the LLK team so they can help out.

**Comment by sjameelTT** [comment]
I believe this should work with https://github.com/tenstorrent/tt-metal/pull/22652

**Comment by nsorabaTT** [comment]
Confirmed that the fix works for the topk issue. Closing.
